src/test/kotlin/coverosR3z/timerecording/CreateEmployeeAPITests.kt

```diff
@@ -110,4 +77,93 @@ class CreateEmployeeAPITests {
         assertEquals("expected keys: [employee_name]. received keys: []", ex.message)
     }
 
+    // region ROLE TESTS
+
+    @Category(APITestCategory::class)
+    @Test
+    fun testShouldAllowAdminForPost() {
+        val data = PostBodyData(mapOf(Elements.EMPLOYEE_INPUT.getElemName() to DEFAULT_EMPLOYEE_NAME.value))
+        val sd = makeServerData(data, tru, au, user = DEFAULT_ADMIN_USER)
+
+        assertEquals(StatusCode.OK, CreateEmployeeAPI.handlePost(sd).statusCode)
+    }
+
+    @Category(APITestCategory::class)
+    @Test
+    fun testShouldAllowSystemForPost() {
+        val data = PostBodyData(mapOf(Elements.EMPLOYEE_INPUT.getElemName() to DEFAULT_EMPLOYEE_NAME.value))
+        val sd = makeServerData(data, tru, au, user = SYSTEM_USER)
+
+        assertEquals(StatusCode.OK, CreateEmployeeAPI.handlePost(sd).statusCode)
+    }
+
+    /**
+     * if a user with an improper role tries something, the exception gets
+     * bubbled up to [coverosR3z.server.utility.ServerUtilities.handleRequest]
+     */
+    @Category(APITestCategory::class)
+    @Test
+    fun testShouldDisallowRegularUserForPost() {
+        val sd = makeServerData(PostBodyData(), tru, au, user = DEFAULT_REGULAR_USER)
+
+        assertThrows(UnpermittedOperationException::class.java) { CreateEmployeeAPI.handlePost(sd) }
+    }
+
+    /**
+     * if a user with an improper role tries something, the exception gets
+     * bubbled up to [coverosR3z.server.utility.ServerUtilities.handleRequest]
+     */
+    @Category(APITestCategory::class)
+    @Test
+    fun testShouldDisallowApproverUserForPost() {
+        val sd = makeServerData(PostBodyData(), tru, au, user = DEFAULT_APPROVER)
+
+        assertThrows(UnpermittedOperationException::class.java) { CreateEmployeeAPI.handlePost(sd) }
+    }
+
+    @Category(APITestCategory::class)
+    @Test
+    fun testShouldAllowAdminToGetPageForPost() {
+        val data = PostBodyData(mapOf(Elements.EMPLOYEE_INPUT.getElemName() to DEFAULT_EMPLOYEE_NAME.value))
+        val sd = makeServerData(data, tru, au, user = DEFAULT_ADMIN_USER)
+
+        assertEquals(StatusCode.OK, CreateEmployeeAPI.handleGet(sd).statusCode)
+    }
+
+    /**
+     * Why would the system be GET'ing this page? disallow
+     */
+    @Category(APITestCategory::class)
+    @Test
+    fun testShouldDisallowSystemToGetPage() {
+        val sd = makeServerData(PostBodyData(), tru, au, user = SYSTEM_USER)
```

Comment:
Isn't user = SYSTEM_USER the default? Also, could we refactor some of these in the same way as LoggingAPITests?

Reply:
yes and yes, and on using SYSTEM_USER, it's just to make it more explicit and legible when reading the test.